modules/core_private/fsw/inc/cfe_core_resourceid_basevalues.h

```diff
@@ -0,0 +1,95 @@
+/*
+**  GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**  Copyright (c) 2006-2019 United States Government as represented by
+**  the Administrator of the National Aeronautics and Space Administration.
+**  All Rights Reserved.
+**
+**  Licensed under the Apache License, Version 2.0 (the "License");
+**  you may not use this file except in compliance with the License.
+**  You may obtain a copy of the License at
+**
+**    http://www.apache.org/licenses/LICENSE-2.0
+**
+**  Unless required by applicable law or agreed to in writing, software
+**  distributed under the License is distributed on an "AS IS" BASIS,
+**  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**  See the License for the specific language governing permissions and
+**  limitations under the License.
+*/
+
+/**
+ * @file
+ *
+ * Contains CFE internal prototypes and definitions related to resource
+ * management and related CFE resource IDs.
+ *
+ * A CFE ES Resource ID is a common way to identify CFE-managed resources such
+ * as apps, tasks, counters, memory pools, CDS blocks, and other entities.
+ */
+
+#ifndef CFE_CORE_RESOURCEID_BASEVALUES_H
+#define CFE_CORE_RESOURCEID_BASEVALUES_H
+
+/*
+** Include Files
+*/
+#include "cfe_resourceid_basevalue.h"
+
+/** @defgroup CFEESResourceIDBase cFE Resource ID base values
+ * @{
+ */
+
+/*
+ * Assign unique offsets per resource types used in CFE core apps.
+ *
+ * Applications should not use these values directly, but rather
+ * in conjuction with the CFE_RESOURCEID_MAKE_BASE macro provided
+ * by the Resource ID module.  (see below)
+ */
+enum
+{
+    /*
+     * Note for Task ID base value --
+     * This currently shares the same offset as OSAL tasks, such that
+     * when "simple" (non-enforcing/backward-compatible) IDs are selected,
+     * the CFE task IDs and the OSAL task IDs end up as the same value.
+     *
+     * The "CFE_RESOURCEID_MARK" bit still differentiates the value when
+     * in strict mode, so there is no overlap in that case.
+     */
+    CFE_RESOURCEID_ES_TASKID_BASE_OFFSET = OS_OBJECT_TYPE_OS_TASK,
+
+    /* Other ES managed resources */
+    CFE_RESOURCEID_ES_APPID_BASE_OFFSET      = OS_OBJECT_TYPE_USER + 1,
+    CFE_RESOURCEID_ES_LIBID_BASE_OFFSET      = OS_OBJECT_TYPE_USER + 2,
+    CFE_RESOURCEID_ES_COUNTID_BASE_OFFSET    = OS_OBJECT_TYPE_USER + 3,
+    CFE_RESOURCEID_ES_POOLID_BASE_OFFSET     = OS_OBJECT_TYPE_USER + 4,
+    CFE_RESOURCEID_ES_CDSBLOCKID_BASE_OFFSET = OS_OBJECT_TYPE_USER + 5,
+
+    /* SB managed resources */
+    CFE_RESOURCEID_SB_PIPEID_RESOURCE_BASE_OFFSET = OS_OBJECT_TYPE_USER + 6
```

Comment:
There seems to be a design decision to abstract cFE resource ID base values and limits by defining cfe_resourceid_basevalue.h using a local mission_build.cmake that would construct cfe_resourceid_basevalue.h using some overridable file. The fallback file for cfe_resourceid_basevalue.h is cfe_resourceid_osal_compatible.h. This file defines CFE_RESOURCE_ID_SHIFT, CFE_RESOURCEID_MAX, and CFE_RESOURCEID_MAKE_BASE to values that involve preprocessor definitions defined in OSAL. The file cfe_core_resourceid_basevalues.h has direct references to OS_OBJECT_TYPE_USER and OS_OBJECT_TYPE_OS_TASK when constructing ofsets for various ES and SB managed resource IDs. Should these offsets also be abstracted in a similar manner? Could we not just directly rely on user-defined CFE_RESOURCEID_MARK and CFE_RESOURCEID_SHIFT to separate the values by calling CFE_RESOURCEID_MAKE_BASE(n) where n is >=0 incrementing?

Reply:
OTOH I was pondering if we should remove the `cfe_resourceid_osal_compatible.h` option and just force use of this config for now.  It is a proof of concept / future work to allow this (if desired) but right now due to other reasons we need to ensure CFE and OSAL IDs do not overlap, so its really not possible to use anything else.

Reply:
See https://github.com/nasa/cFE/issues/1301